### PR TITLE
Use new Celery, use new application pattern

### PR DESCRIPTION
**Pull request by agjohnson**
* Use modern celery
* Drop djcelery
* New pattern for starting celery
* Bump redis to 2.10.6 to avoid startup bug, change autodiscover call

### CODE REVIEW

File: readthedocs/projects/tasks.py [hunk 1]
```diff
@@ -572,7 +569,7 @@ def update_imported_docs(version_pk):
 
 
 # Web tasks
-@task(queue='web')
+@shared_task(queue='web')
```
Comment (safwanrahman) [inline readthedocs/projects/tasks.py]: I dont like using `shared_task` unless its a reusable app where you can not get the app. 
You can import the `app` from `worker` then use the decorator `task`
something like
```python
import readthedocs.worker import app

@app.task(queue='web')
```

Comment (safwanrahman) [reply readthedocs/projects/tasks.py]: https://groups.google.com/forum/#!topic/celery-users/XiSDiNjBR6k

Comment (agjohnson) [reply readthedocs/projects/tasks.py]: Makes sense, I'll try the change.

**Comment by agjohnson** [comment]
Think we've been through this PR a few ways now. I'll get this merged

**Pull request merged**